src/libs/Navigation/AppNavigator/AuthScreens.js

```diff
@@ -131,12 +132,25 @@ class AuthScreens extends React.Component {
         // Load policies, maybe creating a new policy first.
         Linking.getInitialURL()
             .then((url) => {
-                if (this.shouldCreateFreePolicy(url)) {
+                if (!url) {
+                    return;
+                }
+                const path = new URL(url).pathname;
+                const params = new URLSearchParams(url);
+                const exitTo = params.get('exitTo');
+                const email = params.get('email');
+                const isLoggingInAsNewUser = this.props.session && this.props.session.email !== email;
+                const shouldCreateFreePolicy = !isLoggingInAsNewUser
+                    && Str.startsWith(path, Str.normalizeUrl(ROUTES.TRANSITION))
+                    && exitTo === ROUTES.WORKSPACE_NEW;
+                if (shouldCreateFreePolicy) {
                     Policy.createAndGetPolicyList();
                     return;
                 }
-
                 Policy.getPolicyList();
+                if (!isLoggingInAsNewUser && exitTo) {
+                    this.navigateToExitRoute(exitTo);
```

Comment:
When I navigate from the AuthScreens I see the log line `[hmmm] [Navigation] navigate failed because navigation ref was not yet ready - {"route":"workspace/13F98CF7180FA033"}`. I tried to wait for the navigationRef using [onReady()](https://reactnavigation.org/docs/navigation-container/#onready), but it is only "called after the navigation container and all its children finish mounting for the first time". I found this [issue comment](https://github.com/react-navigation/react-navigation/issues/9170#issuecomment-898831026) from @roryabraham who encountered the same thing previously.

I tried using `Linking.openURL` instead which worked but it opened a new tab. It's very strange that we don't see this problem currently when navigating in LogInWithShortLivedTokenPage. Why would that be?https://github.com/Expensify/App/blob/68f0bbf9c7fe8a95c7d65b0a9153229375863d1e/src/pages/LogInWithShortLivedTokenPage.js#L81-L83

If I add a 500ms delay before navigating then it works. The docs say that "A navigator needs to be rendered to be able to handle actions", so is it possible that the navigation runs before the `RootStack.Navigator` is mounted?

@marcaaron what would you recommend?

Reply:
>I tried to wait for the navigationRef using onReady(), but it is only "called after the navigation container and all its children finish mounting for the first time"

How did we try to do this exactly?

>It's very strange that we don't see this problem currently when navigating in LogInWithShortLivedTokenPage

Maybe because we are navigating from inside the page? The navigation ref is probably ready by then.

Reply:
I would recommend we allow navigating before the navigation ref is ready. It seems strange to log that we can't navigate when a better solution would be to just defer the navigation until the ref is ready.

Reply:
> I would recommend we allow navigating before the navigation ref is ready.

Do you mean queuing up the navigation actions and then running them once it is ready?

> How did we try to do this exactly?

I used createOnReadyTask in Navigation.js
```javascript
const [isNavigationReady, setIsNavigationReady] = createOnReadyTask();
```
I exported those functions and called setIsNavigationReady() from [setNavigationReady()](https://github.com/Expensify/App/blob/111cee135a26b5d8492b5f4a974c2cd8cc8f3881/src/Expensify.js#L143) in Expensify.js. I put the navigation in AuthScreens inside `isNavigationReady().then`. The problem is that onReady() is only called once, before the navigation occurs in AuthScreens. So the navigation is ready at first, and then becomes unready later. I would guess because we conditionally render the navigators? Rory had similar problems.

Reply:
>Do you mean queuing up the navigation actions and then running them once it is ready?

Not quite. I meant to do something similar to what you're trying here just set it to ready in the right spot. But `onReady()` doesn't seem like it will work based on what you are saying so that doesn't sound ideal.